perfkitbenchmarker/providers/ibmcloud/ibmcloud.py

```diff
@@ -0,0 +1,833 @@
+
+  _type = 'region'
+
+  def Show(self, region):
+    return self._g.AccountReq('GET', 'regions/%s' % region)
+
+
+class KeyManager(BaseManager):
+  """Handles requests on ssh keys"""
+
+  _type = 'keys'
+
+  def Create(self, key, type, **kwargs):
+    _req_data = {
+      'public_key': key,
+      'type': type
+    }
+    if kwargs.get('name', None):
+      _req_data['name'] = kwargs.get('name', None)
+    if kwargs.get('resource_group', None):
+      _req_data['resource_group'] = {'id':kwargs.get('resource_group', None)}
+    return self._g.create_key(_req_data)
+
+  def List(self, account=None):
+    return self._g.AccountReq('GET', self._type)
+
+  def Show(self, key, account=None):
+    key_id = self.GetUris(key)
+    return self._g.Request('GET', key_id)
+
+  def Delete(self, key, account=None):
+    key_id = self.GetUris(key)
+    return self._g.Request('DELETE', key_id)
+
+
+class NetworkAclManager(BaseManager):
+  """Handles requests on network acls"""
+
+  _type = 'network_acls'
+
+  def Create(self, name):
+    _req_data = {
+      "name": name
+    }
+    return self._g.create_network_acls(_req_data)
+
+  def List(self, account=None):
+    return self._g.AccountReq('GET', 'network_acls')
+
+  def Show(self, network_acl, account=None):
+    network_acl_id = self.GetUris(network_acl)
+    return self._g.Request('GET', network_acl_id)
+
+  def Delete(self, network_acl, account=None):
+    network_acl_id = self.GetUris(network_acl)
+    return self._g.Request('DELETE', network_acl_id)
+
+
+class SubnetManager(BaseManager):
+  """Handles requests on subnets"""
+
+  _type = 'subnet'
+
+  def Create(self, subnet, vpcid, **kwargs):
+    _req_data = {
+      "ipv4_cidr_block": subnet,
+      "vpc": {"id": vpcid},
+      "ip_version": "ipv4",
+    }
+    if kwargs.get('name'):
+      _req_data['name'] = kwargs.get('name')
+    if kwargs.get('zone'):
+      _req_data['zone'] = {
+          'name' : kwargs.get('zone')
+      }
+    return self._g.create_subnet(_req_data)
+
+  def Show(self, subnet, account=None):
+    subnet_id = self.GetUri(subnet)
+    return self._g.Request('GET', subnet_id)
+
+  def Delete(self, subnet, account=None):
+    subnet_id = self.GetUri(subnet)
+    return self._g.Request('DELETE', subnet_id)
+
+  def Patch(self, **kwargs):
+    subnet_id = self.GetUri(kwargs.get('subnet'))
+    return self._g.Request('PATCH', subnet_id, kwargs)
+
+  def ShowPg(self, subnet):
+    uri = self.GetUri(subnet) + '/public_gateway'
+    return self._g.Request('GET', uri)
+
+  def AttachPg(self, subnet, pgid):
+    uri = self.GetUri(subnet) + '/public_gateway'
+    _req_data = {
+       'id': pgid
+        }
+    return self._g.Request('PUT', uri, _req_data)
+
+  def DeletePg(self, subnet):
+    uri = self.GetUri(subnet) + '/public_gateway'
+    return self._g.Request('DELETE', uri)
+
+
+class ImageManager(BaseManager):
+  """Handles requests on images"""
+
+  _type = 'image'
+
+  def Create(self, href, osname, name=None):
```

Comment:
Please add function and class documentation - applies to all functions and classes. See https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/providers/gcp/gce_virtual_machine.py#L133

https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/providers/gcp/gce_virtual_machine.py#L222

as an examples. Without string docs, this will not pass internal linters.

Reply:
yes I will add any missing docs.